pkg/skaffold/schema/defaults/defaults_test.go

```diff
@@ -111,6 +111,27 @@ func TestSetDefaultsOnCluster(t *testing.T) {
 
 		t.CheckNoError(err)
 		t.CheckDeepEqual(constants.DefaultKanikoSecretName, cfg.Build.Cluster.PullSecretName)
+		t.CheckDeepEqual(constants.DefaultKanikoSecretMountPath, cfg.Build.Cluster.PullSecretMountPath)
```

Comment:
Add a test for when the `secret` is optional and make sure `PullSecretMountPath` is not null.

Reply:
By "when the secret is optional" do you mean when `PullSecret` and `PullSecretName` are not set?

Reply:
yes i think that what @prary accomplished in #2910.